src/analysis/normed_space/dual.lean

```diff
@@ -148,11 +145,55 @@ variables {E : Type*} [normed_group E] [normed_space 𝕜 E]
 λ _ _, by simp only [zero_le_one, continuous_linear_map.zero_apply, norm_zero]
 
 lemma polar_eq_Inter (s : set E) :
-  polar 𝕜 s = ⋂ z ∈ s, {x' : dual 𝕜 E | ∥ x' z ∥ ≤ 1} :=
-by { ext, simp only [polar, mem_bInter_iff, mem_set_of_eq], }
+  polar 𝕜 s = ⋂ z ∈ s, {x' : dual 𝕜 E | ∥x' z∥ ≤ 1} :=
+by simp only [polar, set_of_forall]
+
+@[simp] lemma polar_univ : polar 𝕜 (univ : set E) = {0} :=
+begin
+  refine eq_singleton_iff_unique_mem.2 ⟨zero_mem_polar _ _, λ x' hx', _⟩,
+  ext x,
+  refine norm_le_zero_iff.1 (le_of_forall_le_of_dense $ λ ε hε, _),
+  rcases normed_field.exists_norm_lt 𝕜 hε with ⟨c, hc, hcε⟩,
+  calc ∥x' x∥ = ∥c∥ * ∥x' (c⁻¹ • x)∥ :
+    by rw [x'.map_smul, norm_smul, normed_field.norm_inv,
+      mul_inv_cancel_left₀ hc.ne']
+  ... ≤ ε * 1 : mul_le_mul hcε.le (hx' _ trivial) (norm_nonneg _) hε.le
+  ... = ε : mul_one _
+end
+
+lemma is_closed_polar (s : set E) : is_closed (polar 𝕜 s) :=
+begin
+  simp only [polar_eq_Inter, ← continuous_linear_map.apply_apply _ (_ : dual 𝕜 E)],
+  refine is_closed_bInter (λ z hz, _),
+  exact is_closed_Iic.preimage (continuous_linear_map.apply 𝕜 𝕜 z).continuous.norm
+end
+
+variable (E)
+
+lemma polar_gc :
```

Comment:
```suggestion
/-- `polar 𝕜 : set E → set (normed_space.dual 𝕜 E)` forms an order-reversing Galois connection with
a similarly defined map `set (normed_space.dual 𝕜 E) → set E`. We use `order_dual.to_dual` and
`order_dual.of_dual` to express that `polar` is order-reversing. Instead of defining the dual
operation `unpolar s := {x : E | ∀ x' ∈ s, ∥x' x∥ ≤ 1}` we apply `polar 𝕜` again, then pull the set
from the double dual space to the original space using `normed_space.inclusion_in_double_dual`. -/
lemma polar_gc :
```